hal/src/main/native/athena/HAL.cpp

```diff
@@ -285,8 +288,52 @@ size_t HAL_GetSerialNumber(char* buffer, size_t size) {
 }
 
 size_t HAL_GetComments(char* buffer, size_t size) {
-  // TODO: implement file parsing
-  return 0;
+  fs::path path = "/etc/machine-info";
```

Comment:
Recommend caching the result, in case the user calls this from a periodic function.